File: Datum.java
package Solutions;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Datum {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String[] line = reader.readLine().split(" ");

        String[] weekDays = {"Thursday", "Friday", "Saturday", "Sunday", "Monday", "Tuesday", "Wednesday"};
        int[] months = {31,28,31,30,31,30,31,31,30,31,30,31};

        int day = Integer.parseInt(line[0]);

        for(int i=0; i < Integer.parseInt(line[1])-1; i++) {
            day += months[i];
        }
        System.out.println(weekDays[(day%7)-1]);
    }
}

File: c++/Datum.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int months[] = {31,28,31,30,31,30,31,31,30,31,30,31};
    string weekDays[] = {"Wednesday", "Thursday", "Friday", "Saturday", "Sunday", "Monday", "Tuesday"};
    

    int d, m;
    cin >> d >> m;
    int day = d;
    for(int i=0; i < m-1; i++) {
        day += months[i];
    }
    cout << weekDays[day%7];
    return 0;
}